client/state/data-layer/wpcom/read/site/post-email-subscriptions/index.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * External Dependencies
+ */
+import { get, includes, merge } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { subscribeToNewPostEmail, unsubscribeToNewPostEmail, updateNewPostEmailSubscription } from 'state/reader/follows/actions';
+import { errorNotice } from 'state/notices/actions';
+import { getReaderFollowForBlog } from 'state/selectors';
+
+function validateParameters( params ) {
+	// the only valid param is delivery_frequency
+	const frequency = params.deliveryFrequency;
+	const validFrequencies = [
+		'instantly',
+		'daily',
+		'weekly',
+	];
+	if ( includes( validFrequencies, frequency ) ) {
+		return {
+			delivery_frequency: frequency
+		};
+	}
+	return {};
+}
+
+function requestPostEmailSubscription( { dispatch }, action, next ) {
+	dispatch( http( {
+		method: 'POST',
+		path: `/read/site/${ action.payload.blogId }/post_email_subscriptions/new`,
+		body: validateParameters( action.payload ), // have to have an empty body to make wpcom-http happy
+		apiVersion: '1.2',
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+	next( action );
+}
+
+function receivePostEmailSubscription( store, action, next, response ) {
+	// validate that it worked
+	const subscribed = !! ( response && response.subscribed );
+	if ( ! subscribed ) {
+		// shoot. something went wrong.
+		next( unsubscribeToNewPostEmail( action.payload.blogId ) );
+		return;
+	}
+	// pass this on, but tack in the delivery_frequency that we got back from the API
+	next(
+		merge( {}, action, {
+			payload: {
+				delivery_frequency: get( response, [ 'subscription', 'delivery_frequency' ] )
+			}
+		} )
+	);
+}
+
+function receivePostEmailSubscriptionError( { dispatch }, action, next ) {
+	dispatch( errorNotice( translate( 'Sorry, we had a problem subscribing. Please try again.' ) ) );
+	// dispatch an unsubscribe
+	next( unsubscribeToNewPostEmail( action.payload.blogId ) );
+}
+
+function updatePostEmailSubscription( { dispatch, getState }, action, next ) {
```

Comment:
one actually creates a new subscription object (new), optionally without specifying a frequency, and one updates an existing subscription to a new frequency. if you call new without specifying a frequency, we set it to your current default frequency.